File: app/src/main/java/com/example/finalexer3grp2/ViewPagerAdapter.java
package com.example.finalexer3grp2;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class ViewPagerAdapter extends FragmentStateAdapter {

    public ViewPagerAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return new FilmsFragment();
            case 1:
                return new DiaryFragment();
            case 2:
                return new ListsFragment();
            default:
                return new FilmsFragment();
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: app/src/main/java/com/example/finalexer3grp2/ViewPagerAdapter.kt
package com.example.finalexer3grp2
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter

class ViewPagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :
    FragmentStateAdapter(fragmentManager, lifecycle) {

    override fun getItemCount(): Int = 3

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> FilmsFragment()
            1 -> DiaryFragment()
            2 -> ListsFragment()
            else -> FilmsFragment()
        }
    }
}